modules/tracing/mule-tracer-configuration-impl/src/main/java/org/mule/runtime/tracing/level/impl/config/AutoConfigurableTracingLevelConfiguration.java

```diff
@@ -0,0 +1,106 @@
+import static org.mule.runtime.tracer.exporter.config.api.OpenTelemetrySpanExporterConfigurationProperties.MULE_OPEN_TELEMETRY_EXPORTER_ENABLED;
+import static org.mule.runtime.tracing.level.api.config.TracingLevel.MONITORING;
+import static org.mule.runtime.tracing.level.api.config.TracingLevel.valueOf;
+
+import static java.lang.Boolean.parseBoolean;
+import static java.lang.String.format;
+import static java.lang.System.getProperty;
+import static java.util.Collections.synchronizedList;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.tracer.exporter.config.api.SpanExporterConfiguration;
+import org.mule.runtime.tracing.level.api.config.TracingLevel;
+import org.mule.runtime.tracing.level.api.config.TracingLevelConfiguration;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+
+/**
+ * Autoconfiguration of tracing level. Will use MONITORING as the default level if there is not one configured in a config file.
+ */
+public class AutoConfigurableTracingLevelConfiguration implements TracingLevelConfiguration {
+
+  private static final Logger LOGGER = getLogger(AutoConfigurableTracingLevelConfiguration.class);
+
+  @Inject
+  private MuleContext muleContext;
+  private SpanExporterConfiguration spanExporterConfiguration;
+  private TracingLevelConfiguration delegate;
+  private TracingLevel defaultLevel =
+      valueOf(getProperty(MULE_OPEN_TELEMETRY_EXPORTER_DEFAULT_TRACING_LEVEL, MONITORING.toString()).toUpperCase());
+
+  private final List<Consumer<TracingLevelConfiguration>> consumersOnChange = synchronizedList(new ArrayList<>());
+
+  /**
+   * This constructor is needed for injection in the registry.
+   */
+  public AutoConfigurableTracingLevelConfiguration() {}
+
+  public AutoConfigurableTracingLevelConfiguration(TracingLevelConfiguration delegate) {
+    this.delegate = delegate;
+  }
+
+  @Inject
+  public void setSpanExporterConfiguration(SpanExporterConfiguration spanExporterConfiguration) {
+    this.spanExporterConfiguration = spanExporterConfiguration;
+  }
+
+  @Override
+  public TracingLevel getTracingLevel() {
+    return getTracingLevel(false, null);
+  }
+
+  @Override
+  public TracingLevel getTracingLevelOverride(String location) {
+    return getTracingLevel(true, location);
+  }
+
+  public TracingLevel getTracingLevel(boolean isOverride, String location) {
+    try {
+      if (delegate == null) {
+        this.delegate = new FileTracingLevelConfiguration(muleContext);
+        consumersOnChange.forEach(consumer -> this.delegate.onConfigurationChange(consumer));
+      }
+      TracingLevel level = isOverride ? delegate.getTracingLevelOverride(location) : delegate.getTracingLevel();
+      return level != null ? level : defaultLevel;
+    } catch (IllegalArgumentException e) {
```

Comment:
Why is this needed?

Reply:
that expeption is thrown [here](https://github.com/mulesoft/mule/pull/13352/files#diff-2c719d8516c51af6c354e87bbd91995b620bf23cb6cea2b497e973b0dff19d6eR99)